File: AquariumApi.Models/Device/DeviceScheduleTaskAssignment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AquariumApi.Models
{
    [Table("tblDeviceScheduleTaskAssignment")]
    public class DeviceScheduleTaskAssignment
    {
        [Required]
        public int? Id { get; set; }
        public int ScheduleId { get; set; }
        public int TaskId { get; set; }
        public DateTime StartTime { get; set; }
        public TriggerTypes TriggerTypeId { get; set; }
        public int? TriggerTaskId { get; set; }
        public int? TriggerSensorId { get; set; }
        public GpioPinValue TriggerSensorValue { get; set; }
        public bool Repeat { get; set; }
        public int? RepeatInterval { get; set; }
        public DateTime RepeatEndTime { get; set; }
        public string DateConditions { get; set; }


        [ForeignKey("ScheduleId")]
        public DeviceSchedule Schedule { get; set; }
        [ForeignKey("TaskId")]
        public DeviceScheduleTask Task { get; set; }
        [ForeignKey("TriggerSensorId")]
        public DeviceSensor TriggerSensor { get; set; }
        [ForeignKey("TriggerTaskId")]
        public DeviceScheduleTask TriggerTask { get; set; }

        public bool RunsOnDate(DateTime date)
        {
            var day = date.DayOfWeek;
            if (string.IsNullOrEmpty(DateConditions))
                return true;
            if (DateConditions.Length < 7)
                return int.Parse(DateConditions) == date.Day;
            else
            {
                var dayCondition = DateConditions[(int)day];
                var runs = dayCondition.Equals('1');
                return runs;
            }
        }
    }
}

File: AquariumApi.DeviceApi/ClientApp/src/app/models/DeviceScheduleTaskAssignment.ts
import { DeviceSchedule } from "./DeviceSchedule"
import { DeviceScheduleTask } from "./DeviceScheduleTask"
import { DeviceSensor } from "./DeviceSensor"

export class DeviceScheduleTaskAssignment {
    id: number
    scheduleId: number
    taskId: number
    startTime: string
    triggerTypeId: number = 0
    triggerTaskId: number
    triggerSensorId: number
    triggerSensorValue: number
    repeat: boolean
    repeatInterval: number
    repeatEndTime: string

    schedule: DeviceSchedule | null
    task: DeviceScheduleTask | null
    triggerSensor: DeviceSensor | null
    triggerTask: DeviceScheduleTask | null
}
